### PR TITLE
Computations

### PR DESCRIPTION
**Pull request by jsaarimaa**
Added extension methods for the matrices to perform various computational operations.
Updated the README to give minimal descriptions of the functions.

### CODE REVIEW

File: Px.Utils/Operations/MatrixFunction.cs [hunk 1]
```diff
@@ -1,26 +1,90 @@
...
-using System.Numerics;
 
 namespace Px.Utils.Operations
 {
     /// <summary>
     /// Base class for matrix functions, defines some operation that can be applied to a matrix, such as a transformation or a sum
     /// </summary>
     /// <typeparam name="TData">Type of the data values in the matrix</typeparam>
-    public abstract class MatrixFunction<TData>
+    public static class MatrixFunctionExtensions
```
Comment (sakari-malkki) [inline Px.Utils/Operations/MatrixFunction.cs:13]: I suggest renaming the file to match the new class name

**Review by sakari-malkki — COMMENTED** [review]
There's a conflict with the readme file, apart from that looks good

File: Px.Utils/Operations/MatrixFunctionExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            // Copy the original data to the output matrix
+
+            DataIndexer original = new(newMeta, input.Metadata);
+            int ogIndex = 0;
+            do outData[original.CurrentIndex] = input.Data[ogIndex++];
+            while (original.Next());
+
+            return new Matrix<TData>(newMeta, outData);
+        }
+
```
Comment (sakari-malkki) [inline Px.Utils/Operations/MatrixFunctionExtensions.cs:53]: Summaries for this and ApplyRelative below?

**Review by sakari-malkki — DISMISSED** [review]
Left a suggestion for adding summaries for a couple of public functions, but apart from that is good.

File: Px.Utils/Operations/MatrixFunctionExtensions.cs (hunk 2 above)
Comment (jsaarimaa) [reply Px.Utils/Operations/MatrixFunctionExtensions.cs:59]: Added the missing documentation.

**Review by sakari-malkki — APPROVED** [review]

**Pull request merged**